core/server/worker/src/main/java/alluxio/worker/block/reviewer/AllocationCoordinator.java

```diff
@@ -0,0 +1,51 @@
+package alluxio.worker.block.reviewer;
+
+import alluxio.worker.block.BlockMetadataView;
+import alluxio.worker.block.BlockStoreLocation;
+import alluxio.worker.block.allocator.Allocator;
+import alluxio.worker.block.meta.StorageDirView;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+@NotThreadSafe
+public class AllocationCoordinator {
+  public Reviewer mReviewer;
+  public Allocator mAllocator;
+
+  private static AllocationCoordinator sInstance = null;
+
+  private AllocationCoordinator(BlockMetadataView view) {
+    mAllocator = Allocator.Factory.create(view);
+    mReviewer = Reviewer.Factory.create();
+  }
+
+  public static AllocationCoordinator getInstance(BlockMetadataView view) {
+    if (sInstance == null) {
+      sInstance = new AllocationCoordinator(view);
+    }
+    return sInstance;
+  }
+
+  /**
+   * This is currently only used by tests, where the singleton state needs to be reconfigured.
```

Comment:
I feel like making this not a singleton would remove the need for this method too?